### PR TITLE
C4 small improvements

### PR DESCRIPTION
**Pull request by CodeSandwich**
A handful of small improvements, segregated into docs and code.

### CODE REVIEW

File: src/AddressDriver.sol [hunk 1]
```diff
@@ -43,47 +42,47 @@ contract AddressDriver is Managed, ERC2771Context {
 
     /// @notice Calculates the user ID for the message sender
     /// @return userId The user ID
-    function callerUserId() internal view returns (uint256 userId) {
+    function _callerUserId() internal view returns (uint256 userId) {
```
Comment (xmxanuel) [inline src/AddressDriver.sol:45]: :+1:

File: src/Caller.sol [hunk 2]
```diff
@@ -204,6 +204,6 @@ contract Caller is EIP712("Caller", "1"), ERC2771Context(address(this)) {
         returns (bytes memory returnData)
     {
         // Encode the message sender as per ERC-2771
-        return Address.functionCallWithValue(to, abi.encodePacked(data, sender), value);
+        return Address.functionCallWithValue(to, bytes.concat(data, bytes20(sender)), value);
```
Comment (xmxanuel) [inline src/Caller.sol:207]: interesting. Is the concat cheaper?
Why the bytes20 cast also cheaper?

File: src/Drips.sol [hunk 3]
```diff
@@ -288,11 +288,13 @@ abstract contract Drips {
             receivableCycles = toCycle - fromCycle - maxCycles;
             toCycle -= receivableCycles;
         }
-        DripsState storage state = _dripsStorage().states[assetId][userId];
+        mapping(uint32 => AmtDelta) storage amtDeltas =
+            _dripsStorage().states[assetId][userId].amtDeltas;
         for (uint32 cycle = fromCycle; cycle < toCycle; cycle++) {
-            amtPerCycle += state.amtDeltas[cycle].thisCycle;
+            AmtDelta memory amtDelta = amtDeltas[cycle];
```
Comment (xmxanuel) [inline src/Drips.sol:294]: This makes sense

**Review by xmxanuel — APPROVED** [review]
looks good

File: src/Caller.sol (hunk 2 above)
Comment (CodeSandwich) [reply src/Caller.sol:207]: No, it's the same, it's just that `abi.encodePacked` is now deprecated in Solidity in favor of `bytes.concat`.

File: src/Drips.sol (hunk 3 above)
Comment (CodeSandwich) [reply src/Drips.sol:294]: It actually saves a `sload`, I was a little surprised that it's not optimized better. Not all tips from C4 actually make a difference when benched, but this one does.

**Pull request merged**